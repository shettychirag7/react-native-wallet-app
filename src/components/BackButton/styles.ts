import {StyleSheet} from 'react-native';

const styles = StyleSheet.create({
  container: {
    height: 44,
    width: 44,
    borderRadius: 22,
    backgroundColor: '#bdbdbd',
    opacity: 0.8,
    justifyContent: 'center',
    alignItems: 'center',
  },
});

export default styles;
