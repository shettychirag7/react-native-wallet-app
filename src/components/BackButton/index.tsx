import React from 'react';
import {TouchableOpacity, View, ViewProps} from 'react-native';
import Icon from 'react-native-vector-icons/MaterialIcons';

import styles from './styles';

interface BackButtonProps extends ViewProps {
  onPress?: () => void;
}

const BackButton = (props: BackButtonProps) => {
  return (
    <TouchableOpacity {...props}>
      <View style={styles.container}>
        <Icon name="arrow-back" size={26} color={'white'} />
      </View>
    </TouchableOpacity>
  );
};

export default BackButton;
