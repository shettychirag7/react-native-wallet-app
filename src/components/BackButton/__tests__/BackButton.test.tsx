import {fireEvent, render} from '@testing-library/react-native';
import React from 'react';
import BackButton from '..';

describe('component: BackButton', () => {
  it('renders Back Button correctly', () => {
    const {toJSON} = render(<BackButton />);

    expect(toJSON()).toMatchSnapshot();
  });

  it('on press button should execute callback', () => {
    const onPressMock = jest.fn();
    const {getByTestId} = render(
      <BackButton testID="back-button" onPress={onPressMock} />,
    );

    fireEvent.press(getByTestId('back-button'));

    expect(onPressMock).toHaveBeenCalled();
  });
});
