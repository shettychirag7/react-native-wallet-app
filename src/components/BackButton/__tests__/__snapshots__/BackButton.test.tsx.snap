// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component: BackButton renders Back Button correctly 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "opacity": 1,
    }
  }
>
  <View
    style={
      {
        "alignItems": "center",
        "backgroundColor": "#bdbdbd",
        "borderRadius": 22,
        "height": 44,
        "justifyContent": "center",
        "opacity": 0.8,
        "width": 44,
      }
    }
  >
    <Text
      allowFontScaling={false}
      selectable={false}
      style={
        [
          {
            "color": "white",
            "fontSize": 26,
          },
          undefined,
          {
            "fontFamily": "Material Icons",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
          {},
        ]
      }
    >
      î—„
    </Text>
  </View>
</View>
`;
